test_project/tests/test_import_job/test_import_data.py

```diff
@@ -156,3 +158,26 @@ def test_import_data_with_validation_error(existing_artist: Artist):
     job.parse_data()
     job.refresh_from_db()
     assert job.import_status == ImportJob.ImportStatus.INPUT_ERROR
+
+
+@django.test.override_settings(
+    IMPORT_EXPORT_MAX_DATASET_ROWS=1,
+)
+def test_import_create_with_max_rows(
+    new_artist: Artist,
+):
+    """Test import job max dataset rows validation."""
+    import_job: ImportJob = ArtistImportJobFactory(
+        artists=[new_artist],
+        skip_parse_step=True,
+        is_valid_file=False,
+    )
+    expected_error_message = (
+        "Too many rows `2`(Max: 1). Input file may be broken. If it's "
+        "spreadsheet file, please delete empty rows."
+    )
```

Comment:
I'm not sure that hardcoding value is a good practice. Maybe just check `f"Too many rows `2`" in import_job.error_message`? Or the best way is to move error message in a constance and then just reuse in tests